book/src/lib_usage.md

```diff
@@ -0,0 +1,33 @@
+# Modifying Clippy behavior with attributes
+
+In some cases it is possible extend Clippy coverage to include 3rd party libraries.
+At this moment, only one such modification is possible: adding a
+`#[clippy::format_args]` attribute to a macro that supports `format!`-like syntax.
+
+## `#[clippy::format_args]`
+
+This attribute can be added to a macro that supports `format!`-like syntax.
+It tells Clippy that the macro is a formatting macro, and that the arguments to the macro
+should be linted as if they were arguments to `format!`. Any lint that would apply to a
+`format!` call will also apply to the macro call. The macro may have additional arguments
+before the format string, and these will be ignored.
+
+### Example
+
+Note that the `#[clippy::format_args]` is only available in v1.84, and will
```

Comment:
Is this also a concern when your using a create as a dependency that uses this macro? 

This note should probably at the top of this chapter, as it applies to all attributes. Then the attribute documentation should contain the version it was added.

Reply:
It should be fine for dependencies, they're compiled with rustc which doesn't know about specific clippy attributes. I do think it's worth adding a test in an aux-build, assuming aux-build uses rustc and not clippy-driver

We probably don't need to recommend `rustversion`, that would only be needed if the library author wants to run an old clippy on the library itself

Reply:
It's only needed if multiple clippy versions are being run on the workspace containing the attribute. Might be worth noting as an addendum to the section about crate attributes, but not really a thing worth drawing much attention to.